using-timescaledb/reading-data.md

```diff
@@ -336,26 +319,14 @@ value.
 
 ```sql
 SELECT
-  period AS time,
-  coalesce(
-    data_value,
-    (
-      SELECT data_value FROM my_hypertable m WHERE m.time < period AND m.meter_id = t.meter_id ORDER BY time DESC LIMIT 1
-    )
-  ) AS data_value
-FROM
-  generate_series(time_bucket(interval '2 weeks' / 1080, now()-'2 weeks'::interval), now(), '2 weeks'::interval / 1080) AS period
-  LEFT JOIN (
-    SELECT
-      time_bucket('2 weeks'::interval / 1080, time::timestamptz) as btime,
-      meter_id,
-      avg(data_value) AS data_value
-    FROM my_hypertable
-    WHERE
-      time > now() - '2 weeks'::interval
-      AND meter_id IN (1,2,3,4)
-    GROUP BY 1,2
-  ) t ON t.btime = period;
+  time_bucket_gapfill('5 min'::interval, time, now() - '2 weeks'::interval, now()) as time,
+  meter_id,
+  locf(avg(data_value)) AS data_value
+FROM my_hypertable
+WHERE
+  time > now() - '2 weeks'::interval
+  AND meter_id IN (1,2,3,4)
+GROUP BY 1,2
```

Comment:
Given the long example above with gapfilling, which ended just with filling in NULLs, I think it would be better if we stayed with the same example (trades), and also show the new output that this function would show.   

    date    | volume
------------+--------
 2017-09-30 | 11315
 2017-09-29 |  11315
 2017-09-28 |   8216
 2017-09-27 |   5591
 2017-09-26 |   9182
 2017-09-25 |  14359
 2017-09-24 |      9855
 2017-09-23 |   9855
 2017-09-24 |  9855
 2017-09-23 |  9855
 2017-09-22 |   9855


Two other small nits:
- Do you want to order by time desc or asc?
- I think the numerical syntax "GROUP BY 1,2" is less known by a lot of people.  Perhaps change to the column name?

Reply:
The order is DESC which you can also see in the result. I think the numerical syntax is superior because otherwise you have to repeat the expression or use an alias which might give unexpected results if you have columns by the same name. The current version of the document also uses that syntax.

Reply:
changed